File: web/src/api/tenantPackage.ts
import request from './request'

export interface TenantPackage {
  id: number
  name: string
  code: string
  status: number
  sort: number
  remark: string
  max_users: number
  max_storage_mb: number
  max_plugins: number
  created_at: string
  updated_at: string
}

export interface TenantPackageListParams {
  page?: number
  page_size?: number
  name?: string
  status?: number
}

export interface TenantPackageForm {
  name: string
  code: string
  status: number
  sort: number
  remark: string
  max_users: number
  max_storage_mb: number
  max_plugins: number
  menu_ids: number[]
}

export function getTenantPackageList(params: TenantPackageListParams) {
  return request.get('/tenant-packages', { params })
}

export function getAllTenantPackages() {
  return request.get('/tenant-packages/all')
}

export function getTenantPackage(id: number) {
  return request.get(`/tenant-packages/${id}`)
}

export function createTenantPackage(data: TenantPackageForm) {
  return request.post('/tenant-packages', data)
}

export function updateTenantPackage(id: number, data: TenantPackageForm) {
  return request.put(`/tenant-packages/${id}`, data)
}

export function deleteTenantPackage(id: number) {
  return request.delete(`/tenant-packages/${id}`)
}

File: backend/internal/model/tenant_package.go
package model

type TenantPackage struct {
	SnowflakeModel
	Name        string `json:"name" gorm:"size:100;not null"`
	Code        string `json:"code" gorm:"size:100;not null;uniqueIndex"`
	Status      int    `json:"status" gorm:"default:1"`
	Sort        int    `json:"sort" gorm:"default:0"`
	Remark      string `json:"remark" gorm:"size:500"`
	MaxUsers    int    `json:"max_users" gorm:"default:10"`
	MaxStorageMB int   `json:"max_storage_mb" gorm:"default:1024"`
	MaxPlugins  int    `json:"max_plugins" gorm:"default:5"`
}

func (TenantPackage) TableName() string {
	return "tenant_packages"
}

type TenantPackageMenu struct {
	SnowflakeModel
	PackageID int64 `json:"package_id" gorm:"index;not null"`
	MenuID    int64 `json:"menu_id" gorm:"index;not null"`
}

func (TenantPackageMenu) TableName() string {
	return "tenant_package_menus"
}
